packages/stream/solana/StreamClient.ts

```diff
@@ -947,25 +947,45 @@ export default class SolanaStreamClient extends BaseStreamClient {
   }
 
   /**
-   * Utility function that checks whether associated token account for the recipient exists and adds an instruction to add if not
+   * Utility function that checks whether associated token accounts still exist and adds instructions to add them if not
    */
-  private async checkAssociatedTokenAccount(
-    data: CheckAssociatedTokenAccountData,
-    { invoker }: IInteractStreamSolanaExt,
+  private async checkAssociatedTokenAccounts(
+    data: CheckAssociatedTokenAccountsData,
+    { invoker, checkTokenAccounts }: IInteractStreamSolanaExt,
     ixs: TransactionInstruction[]
   ) {
-    const accountExists = await this.connection.getAccountInfo(data.recipientTokens);
-    if (!accountExists?.data) {
-      ixs.push(
-        createAssociatedTokenAccountInstruction(
-          invoker.publicKey!,
-          data.recipientTokens,
-          data.recipient,
-          data.mint,
-          TOKEN_PROGRAM_ID,
-          ASSOCIATED_TOKEN_PROGRAM_ID
-        )
-      );
+    if (!checkTokenAccounts) {
+      return;
+    }
+    const checkedKeys: Set<PublicKey> = new Set();
+    const accountArrays = [
```

Comment:
Let's create a proper object out of this?
We could just have a type and cast data to it.

Much easier then to work with indexes

Reply:
It's easier to read then associating [1] to senderTokens for instance.